File: typescript/transfers.ts
#!/usr/bin/env npx ts-node
/**
 * Transfers Example
 *
 * Transfer USD and spot assets between accounts and wallets.
 *
 * Requires: PRIVATE_KEY environment variable
 */

import { HyperliquidSDK } from 'hyperliquid-sdk';

async function main() {
  const privateKey = process.env.PRIVATE_KEY;
  if (!privateKey) {
    console.log("Set PRIVATE_KEY environment variable");
    console.log("Example: export PRIVATE_KEY='0x...'");
    process.exit(1);
  }

  const sdk = new HyperliquidSDK(undefined, { privateKey });
  console.log(`Wallet: ${sdk.address}`);

  // Transfer USD to another address
  // const result = await sdk.transferUsd(
  //   "0x1234567890123456789012345678901234567890",
  //   10.0
  // );
  // console.log(`USD transfer: ${JSON.stringify(result)}`);

  // Transfer spot asset to another address
  // const result = await sdk.transferSpot(
  //   "PURR",  // token (or token index)
  //   "0x1234567890123456789012345678901234567890",  // destination
  //   100.0    // amount
  // );
  // console.log(`Spot transfer: ${JSON.stringify(result)}`);

  // Transfer from spot wallet to perp wallet (internal)
  // const result = await sdk.transferSpotToPerp(100.0);
  // console.log(`Spot to perp: ${JSON.stringify(result)}`);

  // Transfer from perp wallet to spot wallet (internal)
  // const result = await sdk.transferPerpToSpot(100.0);
  // console.log(`Perp to spot: ${JSON.stringify(result)}`);

  // Send asset (generalized transfer)
  // const result = await sdk.sendAsset(
  //   "USDC",  // token (or token index)
  //   100.0,   // amount
  //   "0x1234567890123456789012345678901234567890"  // destination
  // );
  // console.log(`Send asset: ${JSON.stringify(result)}`);

  console.log("Transfer methods available:");
  console.log("  sdk.transferUsd(destination, amount)");
  console.log("  sdk.transferSpot(token, destination, amount)");
  console.log("  sdk.transferSpotToPerp(amount)");
  console.log("  sdk.transferPerpToSpot(amount)");
  console.log("  sdk.sendAsset(token, amount, destination)");
}

main().catch(console.error);

File: python/transfers.py
"""
Transfers Example

Transfer USD and spot assets between accounts and wallets.

Requires: PRIVATE_KEY environment variable
"""

import os
from hyperliquid_sdk import HyperliquidSDK

private_key = os.environ.get("PRIVATE_KEY")
if not private_key:
    print("Set PRIVATE_KEY environment variable")
    print("Example: export PRIVATE_KEY='0x...'")
    exit(1)

sdk = HyperliquidSDK(private_key=private_key)
print(f"Wallet: {sdk.address}")

# Transfer USD to another address
# result = sdk.transfer_usd(
#     destination="0x1234567890123456789012345678901234567890",
#     amount=10.0
# )
# print(f"USD transfer: {result}")

# Transfer spot asset to another address
# result = sdk.transfer_spot(
#     token="PURR",  # or token index
#     destination="0x1234567890123456789012345678901234567890",
#     amount=100.0
# )
# print(f"Spot transfer: {result}")

# Transfer from spot wallet to perp wallet (internal)
# result = sdk.transfer_spot_to_perp(amount=100.0)
# print(f"Spot to perp: {result}")

# Transfer from perp wallet to spot wallet (internal)
# result = sdk.transfer_perp_to_spot(amount=100.0)
# print(f"Perp to spot: {result}")

# Send asset (generalized transfer)
# result = sdk.send_asset(
#     token="USDC",  # or token index
#     amount="100.0",
#     destination="0x1234567890123456789012345678901234567890"
# )
# print(f"Send asset: {result}")

print("Transfer methods available:")
print("  sdk.transfer_usd(destination, amount)")
print("  sdk.transfer_spot(token, destination, amount)")
print("  sdk.transfer_spot_to_perp(amount)")
print("  sdk.transfer_perp_to_spot(amount)")
print("  sdk.send_asset(token, amount, destination)")
